vcl_templates/www.vcl.erb

```diff
@@ -184,9 +184,20 @@ sub vcl_recv {
 
   # FIXME: Prefer a fallback director if we move to Varnish 3
   if (req.restarts > 2) {
-    set req.backend = F_mirror1;
-    set req.http.host = "<%= config.fetch('provider1_mirror_hostname') %>";
-    set req.http.Fastly-Backend-Name = "mirror1";
+    set req.backend = F_mirrorS3;
+    set req.http.host = "<%= config.fetch('s3_mirror_hostname') %>";
+    set req.http.Fastly-Backend-Name = "mirrorS3";
+
+    if (req.url ~ "^(/.*)?/([^#?\s]+)\.(atom|css|csv|docx|gif|html|ico|ics|jpeg|jpg|js|json|ods|odt|pdf|png)([\?#]+.*)?$") {
```

Comment:
This feels like it could be clearer. As I understand it it's testing:

1. Is this a request for something with an explicit extension? If so add the s3-prefix to the request URL
2. Is this a request for the root. If so add the s3-prefix and 'index.html'
3. Is this a request for something that doesn't have an extension, if so add '.html'

Would be clearer as:

1. is this a request for root: make the url `index.html`
2. does this have an extension: if not add '.html'
3. add the prefix

e.g. 
```
if (req.url ~ "^/?$"){
  set req.url = "/index.html";
} else if (  req.url !~ "\." ){ # i.e. the URL doesn't contain a '.'
  set req.url = "/" + req.url + ".html";
}
set req.url = "<%= config.fetch('s3_mirror_prefix') %>" + req.url
```